net/http/civetweb/civetweb.c

```diff
@@ -31,6 +31,13 @@
 #if defined(__GNUC__) && !defined(_GNU_SOURCE)
 #define _GNU_SOURCE /* for setgroups(), pthread_setname_np() */
 #endif
+/* Removing _GNU_SOURCE for Clang to avoid  
+ * implicit declaration of function 'pthread_setname_np' is invalid in C99 
```

Comment:
Can we not add the relevant `#include` instead? @linev ?

Reply:
I guess, it is a way how GNU works - see:
http://man7.org/linux/man-pages/man3/pthread_setname_np.3.html
Probably, CLANG treats these functions differently.

Reply:
So this means
```
#define _GNU_SOURCE
#include <pthread.h>
```
should be enough to fix the warning. Is it not?

Reply:
As I said - I have no experience with **clang**, probably it fails when  _GNU_SOURCE is defined.
I guess this was a reason why @oshadura makes these changes.

Reply:
_GNU_SOURCE definitions used in other places. One could rather try:

      #if !defined(_GNU_SOURCE)
      #define _GNU_SOURCE /* for setgroups(), pthread_setname_np() */
      #endif

But again - it may not be supported by clang on mac.
And function pthread_setname_np is just used for debugging.